griptape/utils/import_utils.py

```diff
@@ -0,0 +1,38 @@
+from importlib import import_module
+from types import ModuleType
+from typing import Optional
+
+
+INSTALL_MAPPING = {
+    "huggingface_hub": "huggingface-hub",
+    "pinecone": "pinecone-client",
+    "opensearchpy": "opensearch-py",
+    "requests_aws4auth": "requests-aws4auth",
+}
+
+
+def import_optional_dependency(name: str) -> Optional[ModuleType]:
```

Comment:
Should we extend this method with a custom mapping in addition to `INSTALL_MAPPING` (that we can may be rename to `DEFAULT_INSTALL_MAPPING`), so it's more flexible?

Can add an optional param like `install_mapping: Optional[dict] = None` and merge it with:

```python
mapping = DEFAULT_INSTALL_MAPPING

if install_mapping:
    mapping.update(install_mapping)

package_name = mapping.get(name)
```

Reply:
Ignore this comment :)